OpenRA.Game/GrantedConditionReference.cs

```diff
@@ -0,0 +1,47 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenRA
+{
+	public struct GrantedConditionReference : IVariablesReferences
```

Comment:
From the discord discusion: IMO it would be better if we could go straight to a placeholder `GrantsVariable<T>` type and use `GrantsVariable<string>`(? it still wasn't clear to me exactly what type we would use for general conditions) instead of changing the they of all conditions now and then again in a followup PR.

Reply:
Done.